commcare_export/cli.py

```diff
@@ -207,19 +218,99 @@ def _get_checkpoint_manager(args):
         return checkpoint_manager
 
 
+def evaluate_query(env, query):
+    with env:
+        try:
+            lazy_result = query.eval(env)
+            if lazy_result is not None:
+                # evaluate lazy results
+                for r in lazy_result:
+                    list(r) if r else r
+        except requests.exceptions.RequestException as e:
+            if e.response.status_code == 401:
+                print("\nAuthentication failed. Please check your credentials.")
+                return None
+            else:
+                raise
+        except KeyboardInterrupt:
+            print('\nExport aborted')
+            return None
+
+
+def location_type_map(args):
```

Comment:
this should be done lazily as part of the main query evaluation. I also don't think it needs to use a minilinq query, you could just use the api client directly to get the API json output and do the processing in python which would be much less verbose. You could do that by passing the `api_client` into the LocationInfoProvider class:

```python
class LocationInfoProvider:
    def __init__(api_client):
        self._api_client = api_client
        self._location_types = None

    @property
    def location_types(self):
        if self._location_types is None:
            self._location_types = get_location_types(self.api_client)
        return self._location_types

def get_location_types(api_client):
    response = api_client.get('location_type')
    .....
```

Reply:
Much better. Done.